src/compressed_tensors/compressors/sparse_compressors/base.py

```diff
@@ -108,3 +120,24 @@ def decompress(
                     weight_data[param_name] = f.get_tensor(full_name)
             decompressed = self.decompress_weight(weight_data)
             yield weight_name, decompressed
+
+        for uncompressed_param_name, safe_path in uncompressed_params.items():
+            with safe_open(safe_path, framework="pt", device=device) as f:
+                value = f.get_tensor(uncompressed_param_name)
+            yield uncompressed_param_name, value
+
+    @staticmethod
+    def should_compress(name: str, expanded_targets: Optional[Set[str]] = None) -> bool:
+        """
+        Check if a parameter should be compressed
+
+        :param name: name of the parameter
+        :param expanded_targets: set of layer prefixes to compress
+        :return: whether or not the parameter should be compressed
+        """
+        if expanded_targets is None:
+            return name.endswith(".weight")
+
+        return (
+            name.endswith(".weight") and name[: -(len(".weight"))] in expanded_targets
```

Comment:
len(".weight") can be hardcoded to the actual length. Should also add that it comes from len(".weight")

Reply:
Since it's not used anywhere else, I'd like to be more explicit; But I updated the docstring to say this function only returns True for weight parameters.